src/main/java/net/jingles/enchantments/enchants/weapon/Decapitation.java

```diff
@@ -0,0 +1,81 @@
+package net.jingles.enchantments.enchants.weapon;
+
+import net.jingles.enchantments.Enchant;
+import net.jingles.enchantments.enchants.CustomEnchant;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.TextComponent;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.enchantments.EnchantmentTarget;
+import org.bukkit.entity.*;
+import org.bukkit.event.Event;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.SkullMeta;
+
+import java.util.*;
+
+@Enchant(name = "Decapitation", key = "decapitation", levelRequirement = 15, maxLevel = 4, targetItem = EnchantmentTarget.WEAPON,
+        description = "Has a 10% (+ 10% per additional level) chance of decapitating players, creepers, skeletons, wither skeletons, zombies or dragons upon killing them.")
+
+//TODO: Fix description?
+//TODO: Lower max probability chance? Is it too OP?
+
+public class Decapitation extends CustomEnchant {
+
+    private final Map<EntityType, Material> DROPPABLE = new HashMap<>();
+
+    public Decapitation(NamespacedKey key) {
+        super(key);
+        DROPPABLE.put(EntityType.ZOMBIE, Material.ZOMBIE_HEAD);
+        DROPPABLE.put(EntityType.SKELETON, Material.SKELETON_SKULL);
+        DROPPABLE.put(EntityType.WITHER_SKELETON, Material.WITHER_SKELETON_SKULL);
+        DROPPABLE.put(EntityType.CREEPER, Material.CREEPER_HEAD);
+        DROPPABLE.put(EntityType.ENDER_DRAGON, Material.DRAGON_HEAD);
+        DROPPABLE.put(EntityType.PLAYER, Material.PLAYER_HEAD);
+    }
+
+    @Override
+    public boolean conflictsWith(Enchantment enchant) {
+        return false;
+    }
+
+    @Override
+    public boolean canTrigger(Inventory inventory, Event event) {
+        ItemStack weapon = getItem(inventory);
+        return weapon != null && hasEnchantment(weapon);
+    }
+
+
+    @EventHandler
+    public void onEntityDeath(EntityDeathEvent event) {
+
+        if ((event.getEntity().getKiller() == null) || (event.getEntity().getKiller() instanceof Mob)) { return; }
+
+        Player player = event.getEntity().getKiller();
+
+        if (!(canTrigger(player.getInventory(), event)) || !(DROPPABLE.containsKey(event.getEntityType()))) { return; }
+
+        int level = getItem(player.getInventory()).getItemMeta().getEnchantLevel(this);
+        double probability = 10 * level / 100D;
+        if (Math.random() >= probability) return;
+
+        ItemStack droppedHead = new ItemStack(DROPPABLE.get(event.getEntityType()));
+
+        if (event.getEntity() instanceof Player) {
+            SkullMeta skullMeta = (SkullMeta) droppedHead.getItemMeta();
+            skullMeta.setOwningPlayer(((Player) event.getEntity()).getPlayer());
```

Comment:
You don't need to call getPlayer() on the Player lol. That's for getting a Player object from an OfflinePlayer (and only works if they are online so why does it exist? idk)